stan/math/opencl/rev/copy.hpp

```diff
@@ -26,19 +94,17 @@ namespace math {
  * @param src source Eigen matrix
  * @return matrix_cl with a copy of the data in the source matrix
  */
-template <typename T, int R, int C, require_var_t<T>* = nullptr>
-inline matrix_cl<T> to_matrix_cl(const Eigen::Matrix<T, R, C>& src) try {
-  matrix_cl<T> dst(src.rows(), src.cols());
-  if (src.size() == 0) {
-    return dst;
-  }
-  dst.val() = to_matrix_cl(src.val().eval());
-  dst.adj() = to_matrix_cl(src.adj().eval());
-  return dst;
-} catch (const cl::Error& e) {
-  check_opencl_error("copy var Eigen->(OpenCL)", e);
-  matrix_cl<T> dst(src.rows(), src.cols());
-  return dst;
+template <typename T, require_eigen_vt<is_var, T>* = nullptr>
+inline var_value<matrix_cl<value_type_t<value_type_t<T>>>> to_matrix_cl(
+    T& src) {
```

Comment:
Is this supposed to be a `const T&`?

Reply:
We need t be able to update adjoint of `src`. I can see a use case of this accepting rvalues, but than we would need to const cast them to update adjoint. Maybe forwarding reference? Hmm, its getting a bit late here, I will think on this some more on monday.

Reply:
Added const. I forgot we can go around adj() and directly access .vi_.adj_ to avoid getting const on the adjoint.